kerastuner/engine/conditions.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2019 The Keras Tuner Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"HyperParameters logic."
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import abc
+import six
+
+from ..protos import kerastuner_pb2
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Condition(object):
+    """Abstract condition for a conditional hyperparameter.
+
+    Subclasses of this object can be passed to a `HyperParameter` to
+    specify that this condition must be met in order for that hyperparameter
+    to be considered active for the `Trial`.
+
+    Example:
+
+    ```
+    condition = kt.conditions.OneOf('a', 'dnn')
```

Comment:
Consider moving this between the a and b definitions below, and add a comment explaining what the semantics are: does this mean that num_layers would be active if its parent has a value of 'a' or a value of 'dnn'?
Or is 'a' meant to be the parent name?

Reply:
Done, forgot to update this to use the `Parent` name